qiskit/quantum_info/states/stabilizerstate.py

```diff
@@ -266,6 +266,47 @@ def expectation_value(self, oper, qargs=None):
 
         return pauli_phase
 
+    def equiv(self, other):
+        """Return True if the two generating sets generate the same stabilizer group.
+
+        Args:
+            other (StabilizerState): another StabilizerState.
+
+        Returns:
+            bool: True if other has a generating set that generates the same StabilizerState.
+
+        Raises:
+            QiskitError: if other is not a StabilizerState.
+        """
+        if not isinstance(other, StabilizerState):
+            raise QiskitError("Other is not a StabilizerState.")
```

Comment:
```suggestion
            other = StabilizerState(other)
```
This might be my preference, but quantum_info pattern.

Reply:
I actually made this change following Jake's comment here: https://github.com/Qiskit/qiskit-terra/pull/9543#discussion_r1098980885

Reply:
I'd specifically commented above that I didn't think we should do this - Shelly originally had your suggestion and changed it to this.  If you have this conversion, then it can fail in all manner of inconvenient ways as well, which just makes the method unreliable for users.

Reply:
I understand your point of view, but I would like to be consistent. It seems strange to me that we approve of this way in other methods.

Reply:
Yeah, that's a super valid point.  You convinced me - sorry Shelly, would you mind changing it back?

Reply:
I'll change it back - thank you both for the reviews!